### ISSUE TITLE
Validation error shown twice on create account screen

### ISSUE THREAD

**Issue by aniri** [code4romania/stam-acasa#326]
The message `Câmpurile de parolă și confirmare parolă nu sunt identice.` is displayed twice on the screen.

Also, only the one below the input field is removed when the validation issue is fixed. The top one remains there even after updating the input field contents.

![Screenshot from 2020-04-22 20-07-11](https://user-images.githubusercontent.com/1358055/80011858-1e54dd00-84d5-11ea-880a-4dce70ea4529.png)

**Comment by andorremus** [comment]
I'll take this

**Comment by RaduCStefanescu** [comment]
All yours @andorremus

**Comment by andorremus** [comment]
RIght, the errors here are shown at the top and under each individual field, if we remove the ones at the top we get 99% of what we want, the only issue is that the password gets validated if it's not got a special character in as well, but that validation message is not shown which essentially means that you click on Creeaza Cont and it takes you back to the same page without any messages. Any thoughts?

**Comment by RaduCStefanescu** [comment]
We should show the validation message if the user does not input special characters in the password, so it's a no go from me if we just return to the previous page.

**Comment by andorremus** [comment]
<img width="394" alt="image" src="https://user-images.githubusercontent.com/9131872/80273551-bd4b2600-86db-11ea-8eb8-530148d74afd.png">

And lose the validation messages at the top? So we essentially have the validation messages underneath each respective field

PS Just noticed the email address validation message, should change that as well

**Comment by RaduCStefanescu** [comment]
Please, let's keep the messages in Romanian 
You are sure you have the latest changes? I remember changing those some time ago.

**Comment by andorremus** [comment]
<img width="936" alt="image" src="https://user-images.githubusercontent.com/9131872/80273601-4c583e00-86dc-11ea-8bbd-093a9841a6b3.png">

Yep, pulled the latest develop before starting to work

**Comment by andorremus** [comment]
Can you please provide me with the text the email validation should display?

**Comment by RaduCStefanescu** [comment]
You sure?  This is what I have in develop:
<img width="1026" alt="Screenshot 2020-04-25 at 10 29 09" src="https://user-images.githubusercontent.com/15039873/80273968-a9a1be80-86df-11ea-8aea-8b123564e472.png">

**Comment by andorremus** [comment]
You're right, I took it from the repo on github, my local looks the same as yours.

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/9131872/80274067-70b61980-86e0-11ea-9d81-fcbbf2458f19.png">


It's just got the stuff I'm working on as well (email validation message and password validation message)

**Comment by bvizureanu** [comment]
Hello,

This is caused by the presence of a summary of errors in the aspnet razor views 

So basically there is a sum of all the validation errors put o top of the form here: https://github.com/code4romania/stam-acasa/blob/6cb85b619b8de7bb17d6d3dab90019989501fb62/backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/ResetPassword.cshtml#L12 

Each of the field also supports pointing out the validation error for itself as per this element(s):
https://github.com/code4romania/stam-acasa/blob/6cb85b619b8de7bb17d6d3dab90019989501fb62/backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/ResetPassword.cshtml#L27

basically removing the summary (line 12) should do the trick ;)
On a huge form with lots of inputs and validations the summary was found to be quite useful form a UX point of view. In this particular scenario it's kind of redundant..

**Comment by andorremus** [comment]
I agree, the issue that I need to solve after I've removed the summary is to make sure that the password validation message gets shown on the field as well, as currently it's not shown. The message I'm referring to is the one that validates the existence of a special character in the password, which is not shown on the field but is shown on the summary.

If we don't show it, we're going to end up with a form submit that does nothing (it does validate the form but since we don't show the summary anymore it's not going to be clear what went wrong)

Make sense? :)

**Comment by andorremus** [comment]
<img width="392" alt="image" src="https://user-images.githubusercontent.com/9131872/80274485-62b5c800-86e3-11ea-9519-dc320b84560c.png">

This is the one I'm referring to from the summary

**Comment by bvizureanu** [comment]
it makes sense but it looks like a different issue than the one in the initial description

**Comment by andorremus** [comment]
Ok, I’ll stick to this and someone can create another one for this password message not showing

**Comment by adascaliteiradu** [comment]
The message: 'Password must have at least one non alphanumeric character' is a model validation error. There are also other model validation errors that may be displayed in the validation summary (e.g. "The username is already taken"). To solve this issue I think it is enough to change the `asp-validation-summary` from `All` to `ModelOnly`. We must create another issue for backend to translate validation messages.

**Comment by aniri** [comment]
> The message: 'Password must have at least one non alphanumeric character' is a model validation error. There are also other model validation errors that may be displayed in the validation summary (e.g. "The username is already taken"). To solve this issue I think it is enough to change the `asp-validation-summary` from `All` to `ModelOnly`. We must create another issue for backend to translate validation messages.

This makes sense. I guess it would be ok to show the frontend validations only under the fields and the validations that come from backend as a summary at the top of the form. @bvizureanu @andorremus what do you think?

**Comment by andorremus** [comment]
We could do that, but what we could also do in order to enhance usability is to add a regex to the password field in order to show the validation on the front end first, otherwise users could be annoyed by the fact that they only get the alphanumerical message after they submit the form.

What do you think? @aniri

**Comment by aniri** [comment]
But this will mean that we are duplicating the check and the validation messages both in frontend and backend :(

**Comment by andorremus** [comment]
Well the back-end validation would become redundant as it would never be shown because we've validated for it already. I think it's more acceptable to do that than to only tell the user they've not added a special character in the password after they've thought about a proper password for a while (presumably).

**Comment by andorremus** [comment]
A slightly off-topic question, if I see another issue (in my case some text that's not translated) do I have to leave or can I fix it as well?

An example is the fact that the email validation has only default error message ( Adresa de email is not provided) or similar

@RaduCStefanescu @aniri

**Comment by aniri** [comment]
> A slightly off-topic question, if I see another issue (in my case some text that's not translated) do I have to leave or can I fix it as well?
> 
> An example is the fact that the email validation has only default error message ( Adresa de email is not provided) or similar
> 
> @RaduCStefanescu @aniri

@andorremus hey, if it's not covered by some other issue, you can of course fix the translation too! thank you!

**Comment by aniri** [comment]
> Well the back-end validation would become redundant as it would never be shown because we've validated for it already. I think it's more acceptable to do that than to only tell the user they've not added a special character in the password after they've thought about a proper password for a while (presumably).

@bvizureanu @RaduCStefanescu should we move the check for password special characters to the frontend validation?
